test/specs/palindrom.errors.test.js

```diff
@@ -113,11 +113,11 @@ describe('Palindrom', () => {
               //respond with an error
               request.respondWith({
                 status: 400,
-                responseText: 'error'
+                responseText: `{"op": "replace", "path": "/", value: "Custom message"}`
               });
               setTimeout(() => {
-                /* onConnectionError should be called once now */
-                assert(spy.calledOnce);
+                /* onConnectionError should NOT be called now */
+                assert(spy.notCalled);
                 done();
               }, 5);
             }, 5);
```

Comment:
You need to add one more test `should call onConnectionError on HTTP 500 response`, that would be identical to the test that you changed here